### DISCUSSION TITLE
Got different OSM IDs for few states

### DISCUSSION THREAD

**Discussion by bujji1** [osm-search/Nominatim#2853] [General]
I installed Nominatim two days ago and found that a few records got different OSM ids compared to the ones in the Nominatim search. Because of that, the parent id links are broken in the installation.

For example, for OSM_ID id in Nominatim **63512** ( Maine ) I got OSM_ID **316961875** in my installation. Because of this, all the child nodes got impacted, they got parent_id as zero. Like this, there are 11 states got impacted and many children.

I am confused about what went wrong and not sure if it is a bug or something else I am missing.

This count shows the ones whose admin-level nodes got different OSM_IDS. All the children depending on these nodes are impacted due to this.

![image](https://user-images.githubusercontent.com/10221950/199001616-ce0b040b-7626-480e-9476-d23019a41897.png)

Thanks for your advice.

Regards

**Comment by mtmail** [comment]
Nominatim merges administrative relations and nodes that represent the same into one place (database row). So it's expected that the database contains a relation for Maine (https://www.openstreetmap.org/relation/63512) and data from the node (https://www.openstreetmap.org/node/316961875) are copied into the relation. Nominatim uses the `wikidata` value (plus place type, geographic location) to find the pairs.

That would fail if the relation is not imported (or broken) in the database.

Did you import only the United States? If you list the affected US states, are they all at the border of the country? In the past we've seen that countries are cut/extracted from the whole planet data and the borders of the cut are outdated. For example on http://download.geofabrik.de/north-america/us.html you see the orange borders on the map top-right of the page. If that misses only 1km then the relation hasn't got all members (ways) and is incomplete.

First check if the affected US states are at borders. If for example Kansas is also affected then the above theory doesn't work.

**Comment by bujji1** [comment]
Thanks for your response @mtmail . I imported the full planet data. No, Kansas was not affected.  For the USA  all other states are imported with admin_level =4 and class='boundary' and type='administrative'. But American Samoa, Maine, MD got the class as 'place' and type as 'state' and admin level value as 15. Here is the import file that I used.


![image](https://user-images.githubusercontent.com/10221950/199248183-4a151d12-b15b-4618-8efa-d84d6b998a91.png)

```
[
{   "keys" : ["wikipedia", "wikipedia:*", "wikidata"],
    "values" : {
        "" : "extra"
    }
},
{
    "keys" : [":prefix", ":suffix", "name:prefix:", "name:suffix:",
              "name:etymology", "name:signed", "name:botanical", "*:wikidata",
              "addr:street:name", "addr:street:type"],
    "values" : {
        "" : "skip"
    }
},
{
    "keys" : ["ref", "int_ref", "nat_ref", "reg_ref", "loc_ref", "old_ref",
              "iata", "icao", "pcode", "ISO3166-2"],
    "values" : {
        "" : "ref"
    }
},
{
    "keys" : ["name", "name:", "int_name", "int_name:", "nat_name", "nat_name:*",
              "reg_name", "reg_name:", "loc_name", "loc_name:",
              "old_name", "old_name:", "alt_name", "alt_name:", "alt_name_*",
              "official_name", "official_name:", "place_name", "place_name:",
              "short_name", "short_name:*", "brand"],
    "values" : {
        "" : "name"
    }
},
{
    "keys" : ["landuse"],
    "values" : {
        "cemetry" : "skip",
        "" : "fallback,with_name"
    }
},
{
    "keys" : ["boundary"],
    "values" : {
        "place" : "skip",
        "postal_code" : "main",
        "" : "main,with_name"
    }
},
{
    "keys" : ["place"],
    "values" : {
        "" : "main"
    }
},
{
    "keys" : ["country_code", "ISO3166-1", "is_in:country_code", "is_in:country",
              "addr:country", "addr:country_code"],
    "values" : {
        "" : "country"
    }
},
{
    "keys" : ["addr:", "is_in:", "tiger:county"],
    "values" : {
        "" : "address"
    }
},
{
    "keys" : ["postal_code", "postcode", "addr:postcode",
              "tiger:zip_left", "tiger:zip_right"],
    "values" : {
        "" : "postcode"
    }
},
{
    "keys" : ["capital"],
    "values" : {
        "" : "extra"
    }
},
{
    "keys" : [ "amenity"],
    "values" : {
        "no" : "skip",
        "college" : "main",
        "kindergarten" : "main",
        "language_school" : "main",
        "music_school" : "main",
        "school" : "main",
         "university" : "main"
        
    }
}

]
```

**Reply by mtmail** [reply]
I saw the same with Maine and added a missing tag now. At least I think that might have confused Nominatim. https://www.openstreetmap.org/changeset/128396538 
![image](https://user-images.githubusercontent.com/3727288/199525319-a153393e-7d6d-4c0b-b776-5aae6c759d15.png)

**Comment by bujji1** [comment]
Thank You @mtmail for your help on this. It is the same case with Maryland also and also for 10 more states in other countries.
1) Can I get you the missing ones comparing the old data set with the new one?
2) To fix my database, do I have to download the latest OSM PBF file and run the process to fix the data in my database? Could you please advise?

Once again thanks for your help.
Regards
Kiran

**Reply by mtmail** [reply]
By data update I mean I used https://nominatim.org/release-docs/latest/admin/Update/ to apply incremental (daily) updates, in my case `nominatim replication --once`. If you don't already have replication setup and your planet installation is over a month old then it's possible that a new import is faster than trying to catch up with the updates.

In my case I "see" (`select name->'name' from placex where osm_type='R' and osm_id=63512;`) the Maine relation now. What are the other missing states?

I'm not concerned about `parent_place_id` being 0 (see https://github.com/osm-search/Nominatim/issues/2794#issuecomment-1212315022), the address search still works, which is the main purpose of Nominatim.

**Comment by bujji1** [comment]
Thanks for your response again @mtmail . I don't have a replication setup and the data is around 15 days old. I will do the full reload. Also Thanks for giving me the link to the comment. I am using parent_id in my code to build the hierarchy :) As mentioned in the comment I use place_addressline.

I compared the old data with the latest I have and here are the missing OSM_ID records. We have 11 missing records for admin_level 4. These are the ones that impacted most child records.

https://docs.google.com/spreadsheets/d/1ZGRtQyiCRCEeexFHk2OmVCHTFTNOBfB41wJ0SU1lrOQ/edit?usp=sharing

The data is ordered by admin_level. Just a quick note below are the ones for admin_level 4. Remaining all are in the above link excel sheet


<html><body>
<google-sheets-html-origin><style type="text/css"></style>

3338139 | administrative | 4 | Bomet | 192798 | Kenya
-- | -- | -- | -- | -- | --
63512 | administrative | 4 | Maine | 148838 | United States
1315120 | administrative | 4 | Antioquia | 120027 | Colombia
9210589 | administrative | 4 | é³¥å¶¼ | 449220 | Taiwan
9209704 | administrative | 4 | é¦¬éžå¶¼ | 449220 | Taiwan
162112 | administrative | 4 | Maryland | 148838 | United States
153310 | administrative | 4 | Xinjiang | 270056 | China
9211932 | administrative | 4 | è•ƒæ²™å°¾ | 449220 | Taiwan
153292 | administrative | 4 | Tibet | 270056 | China
396705 | administrative | 4 | Caldas | 120027 | Colombia


</body>
</html>
